### PR TITLE
Admin can edit#50

### LINKED ISSUE #50: User Story 22, Admin can edit a user's profile data
```
As an admin user
When I visit a user's profile page ("/admin/users/5")
And I click the link to edit the user's profile data
The same behaviors exist as if I were that user trying to change their own data
Except I am returned to the show page path of
"/admin/users/5" when I am finished
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #56

depends on #71

depends on #51

### LINKED ISSUE #147: Refactor Admin Upgrade method
Make the admin user update method three different methods: enable, disable, and upgrade - these can be called as the controller method on a button like : method: :enable.

### PR DESCRIPTION
**Pull request by danbriechle**
adds admin edit and update routes
adds admin edit and update methods
refactors old update method 
adds correct routes to views 
adds sad path tests to admin edit 

tests passing 100% coverage

closes #50
closes #147

### CODE REVIEW

File: app/views/admin/users/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
+<%= form_for @user, :url => {:action => "update", :id => @user.id } do |f| %>
```
Comment (asmolentzov) [inline app/views/admin/users/edit.html.erb]: Hm you shouldn't need the `:url` hash should you? Those should be the defaults for an edit form.

File: app/controllers/admin/users_controller.rb [hunk 2]
```diff
@@ -37,7 +51,16 @@ def update
     else
       redirect_to admin_users_path
     end
+  end
+
+  def edit
+    @user = User.find(params[:format])
```
Comment (asmolentzov) [inline app/controllers/admin/users_controller.rb]: the `params[:format]` suggests to me that something is wonky here. I think this is related to using the `:url` hash in the edit form below. Are you able to use `params[:id]` or similar?

File: app/views/admin/users/show.html.erb [hunk 3]
```diff
@@ -7,8 +7,8 @@
 
 
 <% if @user_role == 'default' %>
-  <%= link_to 'Edit Profile', profile_edit_path %>
-  <%= link_to 'Upgrade to Merchant', admin_merchant_path(@user, :upgrade => true), method: :put %>
+  <%= link_to 'Edit Profile', admin_edit_path(@user.id)%>
```
Comment (asmolentzov) [inline app/views/admin/users/show.html.erb]: The routes stuff that seems funny above might also be related to this.

File: config/routes.rb [hunk 4]
```diff
@@ -18,20 +18,23 @@
 
   namespace :admin do
     resources :users, only: [:index, :show, :update]
+    patch 'user/enable', to: "users#enable"
```
Comment (asmolentzov) [inline config/routes.rb]: Are these two lines (21 and 22) meant to be nested under users? Currently they are not, they are independent of users, which is why `edit` doesn't inherently have an id attached to it, which I think is causing the routes wonkiness above. Nesting them under `:users` might help. Maybe give it a try and check out rails routes and see if your routes allow for an `:id` then e.g. `/admin/users/:id/enable` and `/admin/users/:id/edit` which would be more standard and probably what you want?

File: config/routes.rb [hunk 5]
```diff
@@ -18,20 +18,23 @@
...
+    get '/edit', to: "users#edit"
     resources :orders, only: [:show, :update]
     resources :users, as: :merchants, only: [:index, :show, :update]
+    patch '/merchant/upgrade', to: "users#upgrade"
```
Comment (asmolentzov) [inline config/routes.rb]: Same thing, this is not nested under `:merchants` though it maybe should be?

File: app/views/admin/users/edit.html.erb (hunk 1 above)
Comment (danbriechle) [reply app/views/admin/users/edit.html.erb]: There not. This form is for editing a user, so it redirects to user associated things not admin associated things.
